### ISSUE TITLE
No support for react-native 0.67 and Gradle 7

### ISSUE THREAD

**Issue by op** [daily-co/react-native-daily-js#23]
React Native 0.67+ uses Gradle 7. Daily won't build with this version of Gradle.

```
Build file 'node_modules/@daily-co/react-native-daily-js/android/build.gradle' line: 23

* What went wrong:
A problem occurred evaluating project ':daily-co_react-native-daily-js'.
> Plugin with id 'maven' not found.
```

The legacy `maven` plugin have been removed.
https://docs.gradle.org/7.0/userguide/upgrading_version_6.html#removal_of_the_legacy_maven_plugin

Keeping Gradle in `android/gradle/wrapper/gradle-wrapper.properties` at eg 6.9.1 works for now.

**Comment by filipi87** [comment]
Thanks @op. We will investigate that.

**Comment by thorbenprimke** [comment]
For anyone running into this, I was able to fix it with patch package. 

```
diff --git a/node_modules/@daily-co/react-native-daily-js/android/build.gradle b/node_modules/@daily-co/react-native-daily-js/android/build.gradle
index b8560e5..c6411ad 100644
--- a/node_modules/@daily-co/react-native-daily-js/android/build.gradle
+++ b/node_modules/@daily-co/react-native-daily-js/android/build.gradle
@@ -20,7 +20,7 @@ def safeExtGet(prop, fallback) {
 }
 
 apply plugin: 'com.android.library'
-apply plugin: 'maven'
+apply plugin: 'maven-publish'
 
 buildscript {
     // The Android Gradle plugin is only required when opening the android folder stand-alone.
@@ -39,7 +39,7 @@ buildscript {
 }
 
 apply plugin: 'com.android.library'
-apply plugin: 'maven'
+apply plugin: 'maven-publish'
 
 android {
     compileSdkVersion safeExtGet('compileSdkVersion', DEFAULT_COMPILE_SDK_VERSION)
@@ -113,7 +113,7 @@ afterEvaluate { project ->
     task androidJavadoc(type: Javadoc) {
         source = android.sourceSets.main.java.srcDirs
         classpath += files(android.bootClasspath)
-        classpath += files(project.getConfigurations().getByName('compile').asList())
+//        classpath += files(project.getConfigurations().getByName('compile').asList())
         include '**/*.java'
     }
 

 ```
 
 @filipi87 - any chance y'all could officially upgrade to the latest gradle? With Expo 45 having been released, more folks will hit this.

**Comment by filipi87** [comment]
Hi @thorbenprimke , we already have an open draft PR for that. Just some more tests to make sure all is working fine. 

I am hoping we can release this fix in the next release of `react-native-daily-js`.

**Comment by thorbenprimke** [comment]
I can confirm that it's no longer an issue in version 0.25. thanks @filipi87 !

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia0.giphy.com%2Fmedia%2FAeWoyE3ZT90YM%2Fgiphy.gif&token=thanks" data-gitmeme-token="thanks"><img src="https://media0.giphy.com/media/AeWoyE3ZT90YM/giphy.gif" title="Created by gitme.me with /thanks"/></a>

**Comment by filipi87** [comment]
Hi, exactly @thorbenprimke , we have released the version 0.25 last week, together with some other improvements. Here is the [release notes](https://github.com/daily-co/react-native-daily-js/releases/tag/react-native-daily-js-2022-06-16-0.25.0) with all the changes.

Thank you all for your patience, and let me know if you need anything.
